base_layer/wallet/src/output_manager_service/storage/database/mod.rs

```diff
@@ -102,140 +102,107 @@ where T: OutputManagerBackend + 'static
         Self { db: Arc::new(db) }
     }
 
-    pub async fn add_unspent_output(&self, output: DbUnblindedOutput) -> Result<(), OutputManagerStorageError> {
+    pub fn add_unspent_output(&self, output: DbUnblindedOutput) -> Result<(), OutputManagerStorageError> {
         let db_clone = self.db.clone();
-        tokio::task::spawn_blocking(move || {
-            db_clone.write(WriteOperation::Insert(DbKeyValuePair::UnspentOutput(
-                output.commitment.clone(),
-                Box::new(output),
-            )))
-        })
-        .await
-        .map_err(|err| OutputManagerStorageError::BlockingTaskSpawnError(err.to_string()))??;
+        db_clone.write(WriteOperation::Insert(DbKeyValuePair::UnspentOutput(
+            output.commitment.clone(),
+            Box::new(output),
+        )))?;
 
         Ok(())
     }
 
-    pub async fn add_unspent_output_with_tx_id(
+    pub fn add_unspent_output_with_tx_id(
         &self,
         tx_id: TxId,
         output: DbUnblindedOutput,
     ) -> Result<(), OutputManagerStorageError> {
         let db_clone = self.db.clone();
-        tokio::task::spawn_blocking(move || {
-            db_clone.write(WriteOperation::Insert(DbKeyValuePair::UnspentOutputWithTxId(
-                output.commitment.clone(),
-                (tx_id, Box::new(output)),
-            )))
-        })
-        .await
-        .map_err(|err| OutputManagerStorageError::BlockingTaskSpawnError(err.to_string()))??;
+        db_clone.write(WriteOperation::Insert(DbKeyValuePair::UnspentOutputWithTxId(
+            output.commitment.clone(),
+            (tx_id, Box::new(output)),
+        )))?;
 
         Ok(())
     }
 
-    pub async fn add_unvalidated_output(
+    pub fn add_unvalidated_output(
         &self,
         tx_id: TxId,
         output: DbUnblindedOutput,
     ) -> Result<(), OutputManagerStorageError> {
         let db_clone = self.db.clone();
-        tokio::task::spawn_blocking(move || db_clone.add_unvalidated_output(output, tx_id))
-            .await
-            .map_err(|err| OutputManagerStorageError::BlockingTaskSpawnError(err.to_string()))??;
+        db_clone.add_unvalidated_output(output, tx_id)?;
 
         Ok(())
     }
 
-    pub async fn add_output_to_be_received(
+    pub fn add_output_to_be_received(
         &self,
         tx_id: TxId,
         output: DbUnblindedOutput,
         coinbase_block_height: Option<u64>,
     ) -> Result<(), OutputManagerStorageError> {
         let db_clone = self.db.clone();
```

Comment:
These clones can come out now. There are an artifact from having to `move` into the spawn.